Providers/Profile.swift

```diff
@@ -626,6 +627,14 @@ open class BrowserProfile: Profile {
         return RustLogins(sqlCipherDatabasePath: sqlCipherDatabasePath, databasePath: databasePath)
     }()
 
+    lazy var firefoxSuggest: RustFirefoxSuggest = {
+        let databasePath = URL(
+            fileURLWithPath: (try! files.getAndEnsureDirectory()),
+            isDirectory: true
+        ).appendingPathComponent("suggest.db").path
```

Comment:
I wonder if we should actually store the database in the caches directory—something like:

```suggestion
        let databasePath = try! FileManager.default.url(
            for: .cachesDirectory,
            in: .userDomainMask,
            appropriateFor: nil,
            create: true
        ).appendingPathComponent("suggest.db").path
```

At first, I was thinking of storing the database in the profile directory, so that if we ever add per-user state—for example, which suggestions a user has dismissed—it could all live in the same database.

But I didn't think too hard about that, and the size of the per-user data is going to be so small (maybe a few hundred KB, depending on how many suggestions a user dismisses?) compared to the downloaded suggestions (10 MB), that it doesn't seem worth it. They also have different durability requirements: per-user state has to be backed up, but the suggestions can be re-downloaded anytime.

(Thank you @0c0w3 for bringing that up for Desktop!)

Reply:
I think that, also, thinking of a per-user in a mobile context is..... not necessary. How many apps do you have on your phone that allow you to switch users, that you actively switch a user on?

Reply:
(I went ahead and switched to using the caches directory, instead of the profile).